tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -44,3 +47,109 @@ def faulty_help_example_rule(linter, help_entry):
     if violations:
         raise RuleError('The following example entry indices do not include the command: {}'.format(
             ' | '.join(violations)))
+
+@help_file_entry_rule
+def faulty_help_example_parameters_rule(linter, help_entry):
+    parser = linter.command_parser
+    violations = []
+
+    for example in linter.get_help_entry_examples(help_entry):
+        max_profile = example.get('max_profile')
+        if max_profile and max_profile != 'latest':
+            logger.warning("\n\tSKIPPING example: {}\n\tas its max profile is {}, instead of latest.".format(example['text'], example['max_profile']))
+            continue
+
+        example_text = example.get('text','')
+        commands = _extract_commands_from_example(example_text)
+        while commands:
+            command = commands.pop()
+            violation, nested_commands = _lint_example_command(command, parser)
+
+            commands.extend(nested_commands)  # append commands that are the source of any arguments
+            if violation:
+                violations.append(violation)
+
+    if violations:
+        num_err = len(violations)
+        violation_str = "\n\n".join(violations)
+        violation_msg = "\n\tThere is a violation:\n{}.".format(violation_str) if num_err == 1 else \
+            "\n\tThere are {} violations:\n{}".format(num_err, violation_str)
+        raise RuleError(violation_msg + "\n\n")
+
+
+### Faulty help example parameters rule helpers
+
+@mock.patch("azure.cli.core.parser.AzCliCommandParser._check_value")
+@mock.patch("argparse.ArgumentParser._get_value")
+@mock.patch("azure.cli.core.parser.AzCliCommandParser.error")
+def _lint_example_command(command, parser, mocked_error_method, mocked_get_value, mocked_check_value):
+    def get_value_side_effect(action, arg_string):
+        return arg_string
+    mocked_error_method.side_effect = SystemExit  # mock call of parser.error so usage won't be printed.
```

Comment:
Please do not use SystemExit as your side effect and create your own exception(or use https://github.com/Azure/azure-cli/blob/dev/tools/automation/cli_linter/util.py#L81)
This stops you from being able to differentiate between the side effect and other possible SystemExits.

Reply:
Makes sense, I will throw a LinterError